### PR TITLE
JAMES-3076 Systematically use 'concat' upon denormalization scheduling

### PR DESCRIPTION
**Pull request by chibenwa**
This ensures the projection is updated if, and only if, the source of
truth is updated.

### CODE REVIEW

**Comment by trantienduchn** [comment]
Can I ask what's the source of truth and what is the projection?

**Comment by chibenwa** [comment]
> Can I ask what's the source of truth and what is the projection?

That is quite orthogonal to this PR.

What we should remember here is "we denormalize = we use concat"

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public void save(MailboxMessage mailboxMessage) throws MailboxException {
         mailboxMapper.findMailboxById(mailboxId);
         ComposedMessageIdWithMetaData composedMessageIdWithMetaData = createMetadataFor(mailboxMessage);
         messageDAO.save(mailboxMessage)
-            .thenMany(Flux.merge(
+            .thenMany(Flux.concat(
                 imapUidDAO.insert(composedMessageIdWithMetaData),
                 messageIdDAO.insert(composedMessageIdWithMetaData)))
```
Comment (trantienduchn) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java]: as I understand the `messageIdDAO` is the source and `imapUidDAO` is the projection. Why saving saves the projection first, then the source? I though it should be in a reverse way

Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java]: It's the other way around: that's not what I intend to fix here anyway.

Cf https://github.com/linagora/james-project/pull/3148/commits/325e79b1e6f72a96798c0f0de153b5ee70fa2c37

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public void save(MailboxMessage mailboxMessage) throws MailboxException {
         mailboxMapper.findMailboxById(mailboxId);
         ComposedMessageIdWithMetaData composedMessageIdWithMetaData = createMetadataFor(mailboxMessage);
         messageDAO.save(mailboxMessage)
-            .thenMany(Flux.merge(
+            .thenMany(Flux.concat(
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java]: I don't really like using `Flux.concat` when we want to express `action1.then(action2)`
What about writing `imapUidDAO.insert(composedMessageIdWithMetaData).then(messageIdDAO.insert(composedMessageIdWithMetaData))` here?

**Comment by chibenwa** [comment]
Sire we can but thats morework & could be done later while working on message inconsistency topic.

**Comment by chibenwa** [comment]
Had been done

**Pull request closed without merging**